agent/envoy_bootstrap/platforminfo/platform_info_collector.go

```diff
@@ -264,8 +265,62 @@ func getEcsContainerSupportedIPFamilies(uri string) string {
 		log.Warnf("Containers info not found in ECS metadata: %v", metadataMap)
 		return ""
 	}
-	// all containers share the same networks
-	containerInfo := containers.([]interface{})[0]
+
+	containersArray := containers.([]interface{})
+	firstContainer := containersArray[0]
+
+	// Determine network mode from the first container as it would be same for all
+	networkMode := getNetworkModeFromContainer(firstContainer)
```

Comment:
For bridge non-pause containers, "Networks" looks like below. 
```
"Networks": [
        {
          "NetworkMode": "container:aecea44d6c6333918344b0a7b381a1965088bafe5fad4dd4007a974b919ad5c1",
          "IPv4Addresses": [
            ""
          ]
        }
      ]
```

So, you cannot use it to get network mode. You must use pause containers for which "Networks" looks like below. 

```
      "Networks": [
        {
          "NetworkMode": "bridge",
          "IPv4Addresses": [
            "172.17.0.8"
          ],
          "IPv6Addresses": [
            "2001:db8:1::242:ac11:8"
          ]
        }
      ]
```

Reply:
Can we crash the Agent for such issues?